exonum/src/runtime/rust/call_context.rs

```diff
@@ -0,0 +1,200 @@
+use exonum_merkledb::{BinaryValue, Fork};
+
+use super::super::{
+    dispatcher::{Dispatcher, Error as DispatcherError, Schema},
+    error::catch_panic,
+    ArtifactId, CallInfo, Caller, ExecutionContext, ExecutionError, InstanceDescriptor, InstanceId,
+    InstanceQuery, InstanceSpec, MethodId, SUPERVISOR_INSTANCE_ID,
+};
+
+/// Context for the executed call.
+///
+/// The call can mean a transaction call, or the `before_commit` hook.
+#[derive(Debug)]
+pub struct CallContext<'a> {
+    /// Underlying execution context.
+    inner: ExecutionContext<'a>,
+    /// ID of the executing service.
+    instance: InstanceDescriptor<'a>,
+}
+
+impl<'a> CallContext<'a> {
+    /// Creates a new transaction context for the specified execution context and the instance
+    /// descriptor.
+    pub(crate) fn new(context: ExecutionContext<'a>, instance: InstanceDescriptor<'a>) -> Self {
+        Self {
+            inner: context,
+            instance,
+        }
+    }
+
+    /// Returns a writable snapshot of the current blockchain state.
+    pub fn fork(&self) -> &Fork {
+        self.inner.fork
+    }
+
+    /// Returns the initiator of the actual transaction execution.
+    pub fn caller(&self) -> &Caller {
+        &self.inner.caller
+    }
+
+    /// Accesses dispatcher information.
+    pub fn dispatcher(&self) -> Schema<&Fork> {
```

Comment:
Since the return value is not `Dispatcher`, I guess `dispatcher_info` (or something similar) will be more suitable name.